terratorch/models/pixel_wise_model.py

```diff
@@ -90,7 +90,11 @@ def _check_for_single_channel_and_squeeze(x):
     def forward(self, x: torch.Tensor, **kwargs) -> ModelOutput:
         """Sequentially pass `x` through model`s encoder, decoder and heads"""
         self.check_input_shape(x)
-        input_size = x.shape[-2:]
+        if isinstance(x, torch.Tensor):
+            input_size = x.shape[-2:]
+        elif isinstance(x, dict):
+            # Multimodal input in passed as dict
+            input_size = list(x.values())[0].shape[-2:]
```

Comment:
I fix it so that we can always expect an image modality as the first modality from the generic dataset (in case of pixel-wise tasks). I add a functionality that you can also pass the input size in the kwargs to be more flexible for custom datasets.